recipes/enchant/all/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+cmake_minimum_required(VERSION 3.2)
+
+project(enchant VERSION "${CONAN_enchant_VERSION}")
+
+set(root "${PROJECT_SOURCE_DIR}/source_subfolder")
+set(src "${root}/src")
+set(bin "${PROJECT_BINARY_DIR}")
+
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup(TARGETS KEEP_RPATHS)
+
+# ---- Config headers ----
+
+include(configure.cmake)
+
+# ---- Setup ----
+
+find_package(glib REQUIRED CONFIG)
+
+include(GenerateExportHeader)
+
+function(target_set_visibility target)
+  set_target_properties(
+      "${target}" PROPERTIES
+      C_VISIBILITY_PRESET hidden
+      CXX_VISIBILITY_PRESET hidden
+      VISIBILITY_INLINES_HIDDEN YES
+  )
+endfunction()
+
+set(providers "")
+macro(add_provider target)
+  add_library("${target}" MODULE ${ARGN})
+  set_property(TARGET "${target}" PROPERTY DEFINE_SYMBOL enchant_module_EXPORTS)
+  target_set_visibility("${target}")
+  target_link_libraries("${target}" PRIVATE enchant)
+  if(NOT providers)
+    generate_export_header("${target}" BASE_NAME enchant_module)
```

Comment:
Why not this instead of patching ?
```
set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Reply:
I don't remember the exact details, but the enchant project's build scripts do some processing to add exported symbols. This patch makes those exported symbols explicit using normal methods. Exporting all symbols is also a hack.